relay/sources/relayd/src/remote_run.rs

```diff
@@ -148,11 +148,16 @@ impl FromStr for Condition {
         let re = Regex::new(condition_regex).unwrap();
 
         if !re.is_match(s) {
-            Err(Error::InvalidCondition(format!(
+            return Err(Error::InvalidCondition(format!(
                 "Wrong condition: '{}', it should match {}",
                 s.to_string(),
                 condition_regex
-            )))
+            )));
+        }
+        if s.len() > 1024 {
```

Comment:
Could you move it before the regex check? It could help prevent a DoS based on the expensive regex check (though unlikely to happen :) )